vdr/vdr.go

```diff
@@ -69,6 +70,57 @@ type VDR struct {
 	eventManager      events.Event
 }
 
+func (r *VDR) DeriveWebDIDDocument(ctx context.Context, baseURL url.URL, nutsDID did.DID) (*did.Document, error) {
+	nutsDIDDocument, _, err := r.Resolver().Resolve(nutsDID, nil)
+	if err != nil {
+		return nil, err
+	}
+	isOwner, err := r.IsOwner(ctx, nutsDID)
+	if err != nil {
+		return nil, err
+	}
+	if !isOwner {
+		log.Logger().
+			WithError(err).
+			WithField(core.LogFieldDID, nutsDID).
+			Info("Tried to derive did:web document from Nuts DID that is not owned by this node")
+		return nil, types.ErrNotFound
+	}
+
+	// Replace did:nuts DIDs with did:web, but only for owned DIDs
```

Comment:
references in `did:nuts` format (`did:nuts:123/serviceEndpoint?type=oauth`) are not according to the did spec. The nuts node needs to implement did dereferencing according to spec if we want to be compatible, and we need to convert the did:nuts references for the did:web document. New issue probably